akka-http2-support/src/test/scala/akka/http/impl/engine/http2/Http2ServerSpec.scala

```diff
@@ -922,6 +934,53 @@ class Http2ServerSpec extends AkkaSpecWithMaterializer("""
       }
     }
 
+    "enforce settings" should {
+
+      "reject new substreams when exceeding SETTINGS_MAX_CONCURRENT_STREAMS" in new TestSetup with RequestResponseProbes {
+        def maxStreams: Int = 16
+        override def settings: ServerSettings = super.settings.mapHttp2Settings(_.withMaxConcurrentStreams(maxStreams))
+        val requestHeaderBlock: ByteString = HPackSpecExamples.C41FirstRequestWithHuffman
+
+        // start as many streams as max concurrent...
+        private val streamIds: IndexedSeq[Int] = (0 until maxStreams).map(id => 1 + id * 2)
+        streamIds.foreach(streamId =>
+          sendHEADERS(streamId, endStream = false, endHeaders = true, requestHeaderBlock)
+        )
+
+        // while we don't exceed the limit, there's silence on the line
+        expectNoBytes()
+
+        // When we exceed the limit, though...
+        val lastValidStreamId = streamIds.max
+        val firstInvalidStreamId = lastValidStreamId + 2
+        sendHEADERS(firstInvalidStreamId, endStream = false, endHeaders = true, requestHeaderBlock)
+        expectRST_STREAM(firstInvalidStreamId, ErrorCode.REFUSED_STREAM)
+      }
+
+      "reject new substreams when exceeding SETTINGS_MAX_CONCURRENT_STREAMS (with closed streams in between)" in new TestSetup with RequestResponseProbes with Http2FrameHpackSupport {
```

Comment:
It would be even more reliable when this test would actually open-and-close some streams rather than just skipping some stream ID's, but this is already :+1: